libminifi/src/capi/api.cpp

```diff
@@ -400,11 +395,19 @@ processor *add_processor_with_linkage(flow *flow, const char *processor_name) {
   return nullptr;
 }
 
-int add_failure_callback(flow *flow, void (*onerror_callback)(const flow_file_record*)) {
+int add_failure_callback(flow *flow, void (*onerror_callback)(flow_file_record*)) {
   ExecutionPlan *plan = static_cast<ExecutionPlan*>(flow->plan);
   return plan->setFailureCallback(onerror_callback) ? 0 : 1;
 }
 
+int set_failure_strategy(flow *flow, int strategy) {
```

Comment:
Why is strategy not typed to an enumeration? Might change the dynamics of the return value but it might make conversions more safe.

Reply:
The reason is pretty simple: I didn't have enough coffee when I wrote this, so forgot that C also supports enums. :) Will replace.